### PR TITLE
Transform files identifiers for sharings

### PR DESCRIPTION
**Pull request by nono**

### CODE REVIEW

File: pkg/sharing/files.go [hunk 1]
```diff
@@ -3,36 +3,77 @@ package sharing
...
+
 // EnsureSharedWithMeDir returns the shared-with-me directory, and create it if
 // it doesn't exist
-func EnsureSharedWithMeDir(inst *instance.Instance) error {
+func EnsureSharedWithMeDir(inst *instance.Instance) (*vfs.DirDoc, error) {
 	fs := inst.VFS()
 	dir, _, err := fs.DirOrFileByID(consts.SharedWithMeDirID)
 	if err != nil {
-		return err
+		return nil, err
```
Comment (Gara64) [inline pkg/sharing/files.go]: A missing sharedWithMe dir returns an `err`, you might add:
`if !couchdb.IsNotFoundError(err) {
  return nil, err
}`

File: pkg/sharing/files.go [hunk 2]
```diff
@@ -3,36 +3,77 @@ package sharing
...
+	for i, val := range random {
+		result[2*i] = val & 0xf
+		result[2*i+1] = val >> 4
+	}
+	return result
+}
+
+// XorID transforms the identifier of a file to a new identifier, in a
+// reversible way: it makes a XOR on the hexadecimal characters
+func XorID(id string, key []byte) string {
```
Comment (Gara64) [inline pkg/sharing/files.go]: If the `dir_id` of a file is `io.cozy.files.root-dir`, actually few characters will be XORed. I'm not sure if this is problematic, though.

File: pkg/sharing/files.go [hunk 3]
```diff
@@ -42,10 +83,32 @@ func EnsureSharedWithMeDir(inst *instance.Instance) error {
...
+	if err != nil {
+		return err
+	}
+	fs := inst.VFS()
+	dir, err := vfs.NewDirDocWithParent(rule.Title, parent, []string{"from-sharing-" + s.SID})
+	if err != nil {
+		return err
+	}
+	if rule.Selector == "" || rule.Selector == "id" || rule.Selector == "_id" {
+		dir.DocID = rule.Values[0]
```
[the remark this answers is not in the document]
Comment (nono) [reply pkg/sharing/files.go]: I want to have one directory inside "Shared with me" for each sharing with files involved, and to create it when the sharing is accepted. We have two cases:
- the sharing is for a folder and all the things inside it
- the files are referenced by a set of documents.

Here (line 107), we are in the first case, and `rules.Values[0]` is the XORed identifier of the shared folder.

File: pkg/sharing/files.go (hunk 2 above)
Comment (nono) [reply pkg/sharing/files.go]: What I aim is to not call XorID for root-dir and trash-dir. I'm not sure about the details of how to do that yet.

File: pkg/sharing/files.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (Gara64) [reply pkg/sharing/files.go]: Ok, I was thinking about a single file sharing, we do not address this case yet?

[the remark this answers is not in the document]
Comment (nono) [reply pkg/sharing/files.go]: Not on the short term

**Pull request merged**